### ISSUE TITLE
(CANCELLED) LTS WG Meeting - 2016-12-12 8:00 PM UTC

### ISSUE THREAD

**Issue by TheAlphaNerd** [nodejs/LTS#170]
## Invited

@nodejs/lts 

## When

🎉 It didn't happen! 🎉

## Where

NOWHERE

### nodejs/node
- crypto: allow adding extra certs to well-known CAs [#9139](https://github.com/nodejs/node/pull/9139)
  - Approval to land on v4 and v6
  - Is this important enough to release v4.8 as the next `Argon` release /cc @sam-github
    - Did you notice I found a good use for the code names there! 🎉
> In closed environments, self-signed or privately signed certificates are
commonly used, and rejected by Node.js since their root CAs are not
well-known. Allow extending the set of well-known compiled-in CAs via
environment, so they can be set as a matter of policy.

- backport util.inspect.custom [#9688](https://github.com/nodejs/node/pull/9688)

> Add a util.inspect.Custom Symbol which can be used to customize util.inspect() output. Providing obj[util.inspect.Custom] works like providing obj.inspect, except that the former allows avoiding name clashes with other inspect() methods.

## Minutes

Yeah no...

**Comment by TheAlphaNerd** [comment]
I do want to mention that the only two things we have on the agenda are approvals for `Semver-minor` backports, and a decision about doing an `argon` minor back to back. All of these should be able to be discussed in an async fashion. 

If people are up for it I would like to suggest we postpone the meeting and use this thread to discuss the issues above instead.

Thoughts?

**Comment by sam-github** [comment]
Nice :-). I think there was a move to drop the "v"s from the versions to get people used to it before the 8.x release.

**Comment by Fishrock123** [comment]
1. I think we should probably wait until the PR to be able to use system CAs/Custom CAs lands?
2. Seems unimportant.

**Comment by TheAlphaNerd** [comment]
@Fishrock123 #9139 has landed 
¯\\\_(ツ)\_/¯

**Comment by Fishrock123** [comment]
Huh? The PR I mean is https://github.com/nodejs/node/pull/8334

**Comment by TheAlphaNerd** [comment]
@sam-github can you please comment on the difference between nodejs/node#9139 and nodejs/node#8334

**Comment by Fishrock123** [comment]
> This discussion is about 9139 not 8334

I don't think they are mutually exclusive. I think if we are going to do a minor in LTS for one we might as well wait and do both.

**Comment by sam-github** [comment]
Unrelated. #8334 is to allow linux distros to not only link against system openssl (which node already supports at configure time), but to also use the system openssl's CA roots instead of the built-in.

I don't see why one would wait for the other.

**Comment by sam-github** [comment]
I'll be looking at 8334 today, though, fwiw.

**Comment by TheAlphaNerd** [comment]
@Fishrock123 I was talking to @sam-github at interactive and it would appear that v4 and v6 are broken for using certain proxies without these CAs, making it difficult to use those version in certain regions.

I almost view it as a bug fix for that scenario

**Comment by sam-github** [comment]
Its no a regression, because its sucked for awhile, but @amarzavery (coincidentally) brought up the difficulties some communites are having using node, I pointed him at my PR. It would be nice to get this out, but if it needs to simmer a bit, I guess they can wait a bit longer for a usable npm.

**Comment by Fishrock123** [comment]
> but if it needs to simmer a bit, I guess they can wait a bit longer for a usable npm.

It will take a while regardless. I don't see what a potential extra couple weeks makes in this case given it'l probably more than a month out to begin with.

**Comment by sam-github** [comment]
Could be. I heard if this doesn't go in now, then it causes a cascade of even longer delays, but I don't understand the release pipeline ATM.

When are patches released? When minors? Why was the last 7.x a 7.2.1 instead of a 7.3.0? When will the next 6.x and 4.x come out? Will they contain only patches, or minors and patches?

**Comment by destroyerofbuilds** [comment]
I'm not sure I see the benefit of waiting for 8334 to land before releasing 9139. We save ourselves from two back-to-back minor releases, but what does that buy us? It's my assumption that the risk of releasing 9139 and 8334 is the same, whether they're bundled together, or released separately in two releases.

Also, 8334 does not solve the self-signed corporate cert issue for corporate Windows users. Waiting on 8334 could have an impact on corporate Windows users with no benefit to them, or to the Node community.

**Comment by sam-github** [comment]
I figured out how the currents work.

For the record, I am not trying to railroad this through. If it doesn't match LTS policy, that's OK. We can wait.

But I do think it is low risk - the code is not invoked if the env var is not set, and high value for a not publically vocal but important subset of node users who are having difficulty, particularly in locked down corporate environments.

**Comment by alexwhitman** [comment]
> high value for a not publically vocal but important subset of node users who are having difficulty, particularly in locked down corporate environments

For what it's worth (and apologies if this isn't the place to weigh in) but I'd like to see it go into 6.x.  We have a growing number of Node.js applications with many of them needing to interact with internal systems using certificates signed by our internal CA.  The simplicity of adding an extra `Environment` line to our systemd unit file template would be hugely beneficial.

**Comment by TheAlphaNerd** [comment]
@alexwhitman fwiw it doesn't appear that there will be an issue get the sign off for it to land. The bigger question is how long it will take to land in particular lines. Following our current procedures it will be able to land in `Boron` faster than `Argon` due to the fact that `Argon` just had a semver minor release.

**Comment by sam-github** [comment]
Is Boron 8.x?

**Comment by TheAlphaNerd** [comment]
`Argon` === 4.x
`Boron` === 6.x

**Comment by destroyerofbuilds** [comment]
@TheAlphaNerd could you provide a link to the policy that discourages back-to-back minor releases of an LTS line? (for future reference)

**Comment by TheAlphaNerd** [comment]
@destroyerofbuilds there is no policy, just a general sentiment... most LTS programs do not do minors at all, so we have tried to space ours out

**Comment by TheAlphaNerd** [comment]
So I'm going to go ahead and officially call this meeting off. 

I've opened https://github.com/nodejs/LTS/issues/171 to discuss coming up with a more defined policy around these releases. I think we can safely wait on voting on the two changes landing... the more pressing issue is deciding release cadence

**Comment by destroyerofbuilds** [comment]
> The bigger question is how long it will take to land in particular lines.

My particular scenario (the reason I've requested it be backported), would make backporting _as-soon-as-possible_, as opposed to later, advantageous to me. However, getting it in period would help with the corporate Windows issue I commented on above. Therefore I'm glad that `it doesn't appear that there will be an issue get the sign off for it to land.`.

**Comment by Fishrock123** [comment]
> fwiw it doesn't appear that there will be an issue get the sign off for it to land.

Err, haven't we done this with all other minors up to this point? We'd be best to message that and not surprise the rest of the group who may not be involved in this particular conversation.

**Comment by amarzavery** [comment]
We have Azure customers using the azure-cli on mac and Linux. By getting #8334 backported to 4.x this will help our Linux customers. I definitely agree that a general solution for Windows Enterprise customers would be extremely helpful, however it would be great if we can unblock some of our customers at the earliest rather than waiting for a long time.

**Comment by TheAlphaNerd** [comment]
@Fishrock123 that comment was meant as I've already talked to a few individuals from LTS and the stakeholders on the issue. As such I was under the impression there was not going to be much push back. apologies if you think I was a bit ahead of myself there.
